File: gamepack/Decompiled/VarBitType.java
public class VarBitType extends DualNode {
   public static byte[][] at;
   public static EvictingDualNodeHashTable VarBitType_cached = new EvictingDualNodeHashTable(64);
   static AbstractArchive VarBitType_archive;
   static final int[] ab = new int[32];
   public int startBit;
   public int endBit;
   public int baseVar;

   static {
      int var0 = 2;

      for(int var1 = 0; var1 < 32; ++var1) {
         ab[var1] = var0 - 1;
         var0 += var0;
      }

   }

   VarBitType() {
   }

   void decode(Packet var1) {
      while(true) {
         int var3 = var1.g1();
         if (var3 == 0) {
            return;
         }

         this.decodeNext(var1, var3);
      }
   }

   void decodeNext(Packet var1, int var2) {
      if (var2 == 1) {
         this.baseVar = var1.cl() * -893309655;
         this.startBit = var1.g1() * -1466027509;
         this.endBit = var1.g1() * 909304255;
      }

   }
}

File: find-names/src/main/java/mapper/identifiers/classes/VarBitType.kt
package mapper.identifiers.classes

import mapper.abstractclasses.IdentityMapper
import mapper.abstractclasses.OrderMapper
import mapper.annotations.DependsOn
import mapper.predicateutilities.and
import mapper.predicateutilities.predicateOf
import mapper.wrappers.ClassWrapper
import mapper.wrappers.InstructionWrapper
import mapper.wrappers.MethodWrapper
import org.objectweb.asm.Opcodes
import org.objectweb.asm.Type.INT_TYPE
import org.objectweb.asm.Type.VOID_TYPE

@DependsOn(DualNode::class)
class VarBitType : IdentityMapper.Class() {
    override val predicate = predicateOf<ClassWrapper> { it.superType == type<DualNode>() }
            .and { it.interfaces.isEmpty() }
            .and { it.instanceFields.size == 3 }
            .and { it.instanceFields.all { it.type == INT_TYPE } }

    class decode : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == VOID_TYPE }
                .and { it.instructions.any { it.opcode == Opcodes.GOTO } }
                .and {it.arguments.count()  == 1}
    }

    class decodeNext : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == VOID_TYPE }
                .and { it.instructions.any { it.opcode == Opcodes.GOTO } }
                .and {it.arguments.count()  == 2}
    }



    @DependsOn(decodeNext::class)
    class baseVar : OrderMapper.InMethod.Field(decodeNext::class, 0) {
        override val predicate = predicateOf<InstructionWrapper> { it.opcode == Opcodes.PUTFIELD && it.fieldType == INT_TYPE }
    }

    @DependsOn(decodeNext::class)
    class startBit : OrderMapper.InMethod.Field(decodeNext::class, 1) {
        override val predicate = predicateOf<InstructionWrapper> { it.opcode == Opcodes.PUTFIELD && it.fieldType == INT_TYPE }
    }

    @DependsOn(decodeNext::class)
    class endBit : OrderMapper.InMethod.Field(decodeNext::class, 2) {
        override val predicate = predicateOf<InstructionWrapper> { it.opcode == Opcodes.PUTFIELD && it.fieldType == INT_TYPE }
    }
}